src/Operation/Division.php

```diff
@@ -112,9 +115,7 @@ public function computeWithoutBcMath(DecimalNumber $a, DecimalNumber $b, $precis
             $b = $b->toMagnitude($maxPrecision);
         }
 
-        $result = $this->integerDivision($a, $b, max($precision, $aPrecision));
-
-        return $result;
+        return $this->integerDivision($a, $b, $precision);
```

Comment:
In the current implementation
- if `$aPrecision > $precision` it uses `$aPrecision` although `$precision` have been requested by user
- if `$bPrecision > $aPrecision` it uses `$aPrecision`

I agree your code seems more valid.